doc/source/docker/quickstart.rst

```diff
@@ -0,0 +1,375 @@
+      [ 3] MLX
+      [ 4] NumPy
+      [ 5] PyTorch
+      [ 6] TensorFlow
+      [ 7] sklearn
+
+
+      : 5
+
+      🔨 Creating Flower project quickstart-docker...
+      🎊 Project creation successful.
+
+      Use the following command to run your project:
+
+            cd quickstart-docker
+            pip install -e .
+            flwr run
+
+      $ cd quickstart-docker
+      $ pip install -e .
+
+#. Create a new Docker bridge network called ``flwr``:
+
+   .. code-block:: bash
+
+      $ docker network create --driver bridge flwr
+
+   User-defined networks, such as ``flwr``, enable IP resolution of container names, a feature
+   absent in the default bridge network. This simplifies quickstart by avoiding the need to
+   determine host IP first.
+
+Step 2: Start the SuperLink
+---------------------------
+
+Open your terminal and run:
+
+.. code-block:: bash
+   :substitutions:
+
+   $ docker run --rm \
+         -p 9091:9091 -p 9092:9092 \
+         --network flwr \
+         --name superlink \
+         --detach \
+         flwr/superlink:|latest_version_docker| --insecure
+
+.. dropdown:: Understand the command
+
+   * ``docker run``: This tells Docker to run a container from an image.
+   * ``--rm``: Remove the container once it is stopped or the command exits.
+   * | ``-p 9091:9091 -p 9092:9092``: Map port ``9091`` and ``9092`` of the container to the same port of
+     | the host machine, allowing you to access the Driver API on ``http://localhost:9091`` and
+     | the Fleet API on ``http://localhost:9092``.
+   * ``--network flwr``: Make the container join the network named ``flwr``.
+   * ``--name superlink``: Assign the name ``superlink`` to the container.
+   * ``--detach``: Run the container in the background, freeing up the terminal.
+   * | :substitution-code:`flwr/superlink:|latest_version_docker|`: The name of the image to be run and the specific
+     | tag of the image. The tag :substitution-code:`|latest_version_docker|` represents a specific version of the image.
+   * | ``--insecure``: This flag tells the container to operate in an insecure mode, allowing
+     | unencrypted communication.
+
+Step 3: Start the SuperNode
+---------------------------
+
+The SuperNode Docker image comes with a pre-installed version of Flower and serves as a base for
+building your own SuperNode image.
+
+#. Create a SuperNode Dockerfile called ``Dockerfile.supernode`` and paste the following code into it:
+
+   .. code-block:: dockerfile
+      :caption: Dockerfile.supernode
+      :substitutions:
+
+      FROM flwr/supernode:|latest_version_docker|
+
+      WORKDIR /app
+      RUN python -m pip install -U --no-cache-dir \
+
+      ENTRYPOINT ["flower-supernode"]
+
+#. Copy the dependencies of the Flower project. These can be found in the ``project`` ``dependencies``
+   section of the ``pyproject.toml`` file. As seen in the example below, to copy the dependencies,
+   you would copy the lines 7-9 in the ``pyproject.toml`` file.
```

Comment:
I agree that this allows us to shorten the quickstart example, but I would avoid it because we are only modifying the pyproject.toml file to make it work with Docker.
If we install the fab dependencies automatically at some point, the flwr package will need to be added again in each pyproject.toml.

Reply:
What we can do is to create a pyproject.toml just for docker that does not contain flwr package. we can create the file it with a simple `sed` command: `sed 's/.*flwr\[simulation\].*//' pyproject.toml > docker_pyproject.toml`. wdyt?

Reply:
I have implemented the change in: https://github.com/adap/flower/pull/3722/commits/c355b6fb6f8d875d0d6d59196c6909c3d7d0aff6
I think that was a good idea to use the `pyproject.toml`. Let me know what you think

Reply:
Yes, that looks alright to me too. Thanks!